runtime/container_solaris.go

```diff
@@ -1,19 +1,47 @@
 package runtime
 
 import (
-	"errors"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
 
 	"github.com/docker/containerd/specs"
+	ocs "github.com/opencontainers/specs/specs-go"
 )
 
+func getRootIDs(s *specs.Spec) (int, int, error) {
+	return 0, 0, nil
+}
+
 func (c *container) OOM() (OOM, error) {
-	return nil, errors.New("runtime OOM() not implemented on Solaris")
+	return nil, nil
 }
 
-func (c *container) UpdateResources(r *Resource) error {
-	return errors.New("runtime UpdateResources() not implemented on Solaris")
+func (c *container) Pids() ([]int, error) {
+	var pids []int
+
+	//we get this information from runz state
+	cmd := exec.Command(c.runtime, "state", c.id)
```

Comment:
Ideally `runz` would support the exact same argument format as `runc` even if behind the scene it relies on the `state` file mentioned here.

The goal being to avoid having arch dependent code as much as possible in `containerd` outside some monitoring code as those are not relying on the runtime.